### PR TITLE
Add example with custom query and one-to-many

### PR DESCRIPTION
**Pull request by radovanradic**
There was already example for `@ManyToOne` https://micronaut-projects.github.io/micronaut-data/latest/guide/#dbcNativeQueries and this is adding `@OneToMany` relation example and custom query

### CODE REVIEW

File: src/main/docs/guide/dbc/dbcNativeQueries.adoc [hunk 1]
```diff
@@ -12,8 +12,13 @@ When writing an explicit SQL query if you specify any joins within the query you
...
 For example:
 
-snippet::example.ProductRepository[project-base="doc-examples/jdbc-example", source="main",tags="native"]
+snippet::example.ProductRepository[project-base="doc-examples/jdbc-example", source="main", tags="native"]
 
 In the above example the query uses an alias called `m_` to query the `manufacturer` table via an `INNER JOIN`. Since the returned `Product` entity features a `manufacturer` association it may be nice to materialize this object as well. The `alias` member of the ann:data.annotation.Join[] annotation is used to specify which alias to materialize the `Manufacturer` instance from.
 
 NOTE: It is necessary to use the "logical name" of the field in the `@Join` (the name used in the `@Entity` class) and not the name used in the native query itself. In the previous example, if the name in the class were `myManufacturer`, then you would need to use `Join(value = "myManufacturer", alias = "m_")` without modifying anything on the native sql query.
+
+The above example was many-to-one relation and here is an example of custom query with one-to-many relation:
```
Comment (graemerocher) [inline src/main/docs/guide/dbc/dbcNativeQueries.adoc]:
```suggestion
The above example demonstrated a custom query with a many-to-one relation whilst the following example defines a custom query with one-to-many relation:
```

File: doc-examples/jdbc-example-kotlin/src/main/kotlin/example/BookRepository.kt [hunk 2]
```diff
@@ -124,6 +125,17 @@ interface BookRepository : CrudRepository<Book, Long> { // <2>
...
     // end::procedure[]
 
+    // tag::onetomanycustom[]
+    @Query("""
+        SELECT b.*,
+               r.id AS reviews_id, r.reviewer AS reviews_reviewer, r.content AS reviews_content, r.book_id AS reviews_book_id
+        FROM book b INNER JOIN review r ON b.id = r.book_id
+        WHERE b.title = :title
+        """)
+    @Join("reviews")
```
Comment (graemerocher) [inline doc-examples/jdbc-example-kotlin/src/main/kotlin/example/BookRepository.kt:136]: how does one customise the separator? Looks like it defaults to underscore `_`? Can we document that?

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: doc-examples/jdbc-example-kotlin/src/main/kotlin/example/BookRepository.kt (hunk 2 above)
Comment (radovanradic) [reply doc-examples/jdbc-example-kotlin/src/main/kotlin/example/BookRepository.kt:136]: Added note about that.

File: src/main/docs/guide/dbc/dbcNativeQueries.adoc (hunk 1 above)
Comment (radovanradic) [reply src/main/docs/guide/dbc/dbcNativeQueries.adoc]: Changed.

**Review by graemerocher — APPROVED** [review]

**Review by dstepanov — APPROVED** [review]

**Pull request merged**